app/coffee/CompileController.coffee

```diff
@@ -16,6 +16,8 @@ module.exports = CompileController =
 			ProjectPersistenceManager.markProjectAsJustAccessed request.project_id, (error) ->
 				return next(error) if error?
 				CompileManager.doCompileWithLock request, (error, outputFiles = []) ->
+					unless outputFiles
```

Comment:
are you sure this is needed? `outputFiles = []` above should cover it.

Reply:
It seems to only cover it if `outputFiles` is not specified. If the caller explicitly sets `outputFiles` to `null` as happens in the tests [1], [2] then the default appears not to apply - at least not in node 8 or 10. 

[1] https://github.com/sharelatex/clsi-sharelatex/blob/master/test/unit/coffee/CompileControllerTests.coffee#L87
[2] https://github.com/sharelatex/clsi-sharelatex/blob/master/test/unit/coffee/CompileControllerTests.coffee#L105

Reply:
Originally I was going to remove `null` from the tests and leave that argument unspecified, but sleeping on it I decided that it was safer to check in `doCompileWithLock` in case this behaviour is relied upon in the running code.

Reply:
The default args should cover null as well, if you remove this unless do the tests fail?

```
p = (a = 1) ->
  console.log(a)

p(null)
p(undefined)
p()
p(2)
```

Reply:
Yes, the tests fail consistently both on my laptop and builds.sharelatex.com with
```
  269 passing (22s)
  2 failing

  1) CompileController
       compile
         with an error
           "before each" hook for "should return the JSON response with the error":
     TypeError: Cannot read property 'map' of null
      at app/js/CompileController.js:92:42
      at callCallback (node_modules/sinon/lib/sinon/stub.js:143:22)
      at Object.functionStub (node_modules/sinon/lib/sinon/stub.js:170:21)
      at Function.invoke (node_modules/sinon/lib/sinon/spy.js:342:55)
      at Object.proxy [as doCompileWithLock] (node_modules/sinon/lib/sinon/spy.js:271:26)
      at app/js/CompileController.js:34:33
      at callCallback (node_modules/sinon/lib/sinon/stub.js:143:22)
      at Object.functionStub (node_modules/sinon/lib/sinon/stub.js:170:21)
      at Function.invoke (node_modules/sinon/lib/sinon/spy.js:342:55)
      at Object.proxy [as markProjectAsJustAccessed] (node_modules/sinon/lib/sinon/spy.js:271:26)

  2) CompileController
       compile
         when the request times out
           "before each" hook for "should return the JSON response with the timeout status":
     TypeError: Cannot read property 'map' of null
      at app/js/CompileController.js:92:42
      at callCallback (node_modules/sinon/lib/sinon/stub.js:143:22)
      at Object.functionStub (node_modules/sinon/lib/sinon/stub.js:170:21)
      at Function.invoke (node_modules/sinon/lib/sinon/spy.js:342:55)
      at Object.proxy [as doCompileWithLock] (node_modules/sinon/lib/sinon/spy.js:271:26)
      at app/js/CompileController.js:34:33
      at callCallback (node_modules/sinon/lib/sinon/stub.js:143:22)
      at Object.functionStub (node_modules/sinon/lib/sinon/stub.js:170:21)
      at Function.invoke (node_modules/sinon/lib/sinon/spy.js:342:55)
      at Object.proxy [as markProjectAsJustAccessed] (node_modules/sinon/lib/sinon/spy.js:271:26)



npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! node-clsi@0.1.4 test:unit:_run: `mocha --recursive --reporter spec --exit $@ test/unit/js`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the node-clsi@0.1.4 test:unit:_run script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/node/.npm/_logs/2018-12-17T20_12_09_419Z-debug.log
Makefile:28: recipe for target 'test_unit' failed
make: *** [test_unit] Error 2
```
You can see the full output here: http://builds.sharelatex.com/job/clsi-sharelatex/job/csh-issue-1309-upgrade/2/console

I spent a couple of hours staring at this last night, and came to the conclusion that the errors were caused by `outputFiles` being null. A quick test confirmed that node 6 and 8 both ignore default parameters if the argument is specified as null, and that `null.map` generates an error, so the mystery to me is why this ever worked in the first place.